node/src/main/scala/coop/rchain/node/encode/JsonEncoder.scala

```diff
@@ -9,12 +9,20 @@ import coop.rchain.models.Expr.ExprInstance
 import coop.rchain.models.GUnforgeable.UnfInstance
 import coop.rchain.models.Var.{VarInstance, WildcardMsg}
 import coop.rchain.models._
+import scodec.SizeBound.unknown
+import scodec.bits.BitVector
+import io.circe.parser._
```

Comment:
To make https://github.com/rchain/rchain/pull/3433#discussion_r675856124 perfect

Reply:
I see, this is because we are storing JSON in KV store.

It would be much better if we have scodecs defined directly without going to JSON especially with fields specified as plain strings.
https://github.com/rchain/rchain/blob/1c4572f70da4109bd777b406e297cc5504d4a2de/node/src/main/scala/coop/rchain/node/revvaultexport/reporting/Transaction.scala#L71-L77

---

Do we need two TransactionStore types? Also keys should be typed and not just strings.
https://github.com/rchain/rchain/blob/945bbc9f03d21c077d6dc5922990d8237add9eb6/node/src/main/scala/coop/rchain/node/web/Transaction.scala#L179

https://github.com/rchain/rchain/blob/1c4572f70da4109bd777b406e297cc5504d4a2de/node/src/main/scala/coop/rchain/node/revvaultexport/reporting/Transaction.scala#L26

Reply:
>Do we need two TransactionStore types? 

The second one is the python transaction server store. We will remove this one later. Let me create an issue.https://github.com/rchain/rchain/issues/3490.

> It would be much better if we have scodecs defined directly without going to JSON especially with fields specified as plain strings.

Actually I want it to be more like general encoding like json and protobuf which other languages can support.

Reply:
> The second one is the python transaction server store. We will remove this one later. Let me create an issue.#3490.

I see, this is basically to read externally created LMDB database.

> Actually I want it to be more like general encoding like json and protobuf which other languages can support.

Do you think that we need more general format, it's internal database used by RNode?

I'd be more for protobuf because JSON is slower. But protobuf requires textual proto files with code generation, which makes it much harder to maintain.

Reply:
I fall back to scodec now